netlogo-core/src/main/agent/Turtle.java

```diff
@@ -871,34 +871,34 @@ public void ycor(Double ycor)
 
   public void xandycor(double xcor, double ycor)
       throws AgentException {
-    double oldX = this.xcor;
-    double oldY = this.ycor;
-    xandycorHelper(xcor, ycor);
-    if (_world.tieManager().hasTies()) {
-      _world.tieManager().turtleMoved(this, xcor, ycor, oldX, oldY);
-    }
+    xandycorHelper(xcor, ycor, false);
   }
 
   // when we're calculating ties this gets called since we are recursing
   // and we need to pass the seenTurtles around ev 7/24/07
-  public void xandycor(double xcor, double ycor, scala.collection.immutable.Set<Turtle> seenTurtles)
+  public void xandycor(double xcor, double ycor, scala.collection.immutable.Set<Turtle> seenTurtles, boolean isJump)
       throws AgentException {
-    double oldX = this.xcor;
-    double oldY = this.ycor;
-    xandycorHelper(xcor, ycor);
-    if (_world.tieManager().hasTies()) {
-      _world.tieManager().turtleMoved(this, xcor, ycor, oldX, oldY, seenTurtles);
-    }
+    xandycorHelper(xcor, ycor, seenTurtles, false);
   }
 
-  public void xandycorHelper(double xcor, double ycor)
+  public void xandycorHelper(double xcor, double ycor, boolean isJump)
+    throws AgentException {
+    xandycorHelper(xcor, ycor, Agent$.MODULE$.turtleSet(this), isJump);
+  }
+
+  public void xandycorHelper(double xcor, double ycor, scala.collection.immutable.Set<Turtle> seenTurtles, boolean isJump)
       throws AgentException {
+    double oldX = this.xcor;
+    double oldY = this.ycor;
+
     Patch originalPatch = getPatchHere();
 
     double newX = _world.wrapX(xcor);
     double newY = _world.wrapY(ycor);
 
-    drawLine(this.xcor, this.ycor, xcor, ycor);
+    if (! isJump) {
+      drawLine(this.xcor, this.ycor, xcor, ycor);
+    }
```

Comment:
If I'm reading this right, this skips drawing if we're jumping, but my understanding is that we're supposed to draw a line whenever we move (per the dictionary).  If that's true, I don't think we need any of the `isJump`-specific logic.

Reply:
The arrangement of things here is a bit strange, but we do still draw a line even when jumping. Whereas this method uses [drawLine with start and dest](https://github.com/NetLogo/NetLogo/pull/1509/files#diff-e44dd5e0b96bd9fd2b33db4c98d47d26R110) to draw the line, jumping uses [drawLine with start and distance](https://github.com/NetLogo/NetLogo/pull/1509/files#diff-e44dd5e0b96bd9fd2b33db4c98d47d26R94). We need the separate `drawLine` methods for exact reproducibility as `PenLineMaker` needs to know distance and heading to do wrapping appropriately. Distance and heading are, unfortunately unavailable in the `xandycorHelper` method.

Reply:
Thank you for the explanation, I think I'm starting to understand.  One more question, is [the `isJump` passed in here used](https://github.com/NetLogo/NetLogo/pull/1509/files#diff-5c1f0f72bba41ac7187d99ade4e7e672R879)? I don't see that value actually getting passed to the `xandycorHelper`, and I'm not clear if it's supposed to be ignored.

Reply:
Nice catch! That method is called by `TieManager`, so the `isJump` parameter should probably be removed because tie-based movement is never considered a jump.

Reply:
I've run the drawing tests in Tortoise against this branch and we are aligned, no errors!  Thank you again for this work.